lib/network/kmip/Client.js

```diff
@@ -71,19 +91,19 @@ function _negotiateProtocolVersion(client, logger, cb) {
             logger.error('KMIP::negotiateProtocolVersion',
                          { error: err,
                            vendorIdentification: client.vendorIdentification });
-            return cb(err);
+            return cb(_arsenalError(err));
         }
         const majorVersions =
               response.lookup(searchFilter.protocolVersionMajor);
         const minorVersions =
               response.lookup(searchFilter.protocolVersionMinor);
         if (majorVersions.length === 0 ||
             majorVersions.length !== minorVersions.length) {
-            const error = Error('No suitable protocol version');
+            const error = 'No suitable protocol version';
             logger.error('KMIP::negotiateProtocolVersion',
                          { error,
                            vendorIdentification: client.vendorIdentification });
-            return cb(error);
+            return cb(_arsenalError(error));
```

Comment:
May be convert to _arsenalError and then log. Otherwise we will be logging a string sometimes and sometimes an object for the same property `error`.